pkg/policy/api/selector_test.go

```diff
@@ -38,3 +41,46 @@ func (s *PolicyAPITestSuite) TestSelectsAllEndpoints(c *C) {
 	selectorSlice = EndpointSelectorSlice{NewESFromLabels(labels.ParseSelectLabel("bar")), NewESFromLabels(labels.ParseSelectLabel("foo"))}
 	c.Assert(selectorSlice.SelectsAllEndpoints(), Equals, false)
 }
+
+func benchmarkMatchesSetup(match string, count int) (EndpointSelector, labels.LabelArray) {
+	stringLabels := []string{}
+	for i := 0; i < count; i++ {
+		stringLabels = append(stringLabels, fmt.Sprintf("%d", i))
+	}
+	lbls := labels.NewLabelsFromModel(stringLabels)
+	return NewESFromLabels(lbls.ToSlice()...), labels.ParseLabelArray(match)
+}
+
+func BenchmarkMatchesValid1000(b *testing.B) {
+	es, match := benchmarkMatchesSetup("42", 1000)
```

Comment:
IIRC I think `testing.B` contains a count so you can compare the results for different number of Labels.

Reply:
@aanm after discussion with @joestringer  I have taken over this issue and am using https://github.com/cilium/cilium/pull/4790 to manage it, can we move discussion there?

Reply:
Do you mean `b.N`, used a few lines below?

I think there's two possible benchmarks to do here: Set a fixed number of labels and benchmark using the number of matches, or set a fixed number of matches and benchmark the number of labels. These tests do the former, using `b.N` to count the number of matches. Maybe 1000 labels sounds over the top, but we were observing this bottleneck in live environments so the fact that this benchmark reproduces the issue is useful.